### ISSUE TITLE
Reduce memory usage of txpool_content call

### ISSUE THREAD

**Issue by vkgnosis** [cowprotocol/services#140]
This has (likely) caused pods to get killed because the node returns a large number of transactions in the mempool. The code is in `pending_transaction.rs`.

We could get rid of this approach alltogether if we think it is not that important to recover the previous tx seeing as pod restarts are rare. 

Or make the code handle the response in a streaming way so that it never needs to keep all of it in memory. We know that we are looking only for a specific address,nonce combo which we can search for on the fly. Still it doesn't feel right to have to download so much data from the node for this.

Another alternative is persisting the previous tx data in a database or filesystem so that we do not have to ask the node for it. Or maybe there is a node api that only gives txpool content of a specific address?

**Comment by vkgnosis** [comment]
I also noticed that these requests time out from time to time. It feels more and more that fetching the whole tx pool is overkill for this problem. I'd rather store the tx information somewhere. Having some persistent data for the driver doesn't seem wrong anyway.

**Comment by sunce86** [comment]
Unfortunately, txpool_content is the only way to get the transaction from the tx pool, even if you have the hash or address (AFAIK).
The only performant way would be to query pending transactions from time to time, in parallel thread. 

Another issue with this functionality is that it sometimes takes ~10s to execute.
